### ISSUE TITLE
Package names...

### ISSUE THREAD

**Issue by russplaysguitar** [foundrycf/foundry#9]
So how exactly will fpm know packages by name? Will packages be registered by name at the fpm website?

Assuming the answer to the above is "yes", what if you want to include a dependency for something that is not on the fpm site (such as something written in Javascript)?  Could you maybe just specify a github url? Or github username and package name? Maybe the specific git revision could optionally be included too?

Then... what if you want to specify a dependency for something in a different version control system? Or an arbitrary file hosted somewhere?

**Comment by jfrux** [comment]
With npm, you can specify dependencies with a git URL.
That won't be as reliable of course.

Yes the name is the ID of the package.
Semantic Versioning is also important.

**Comment by russplaysguitar** [comment]
If IDs are package names, will there be a way to specify a dependency on a fork of a project?

**Comment by jfrux** [comment]
Yes, the git URL would override the registry lookup and the name.

The name you provide will dictate the folder it dumps into inside of
foundry_modules.

Sent from my iPhone

On Sep 25, 2012, at 4:59 PM, Russ notifications@github.com wrote:

If IDs are package names, will there be a way to specify a dependency on a
fork of a project?

—
Reply to this email directly or view it on
GitHubhttps://github.com/foundrycf/foundry/issues/9#issuecomment-8870058.
